compiler/resolution/preFold.cpp

```diff
@@ -2151,6 +2151,29 @@ static Expr* unrollHetTupleLoop(CallExpr* call, Expr* tupExpr, Type* iterType) {
   return noop;
 }
 
+// Returns true if 'actual' is defined by:
+//   move( actual, call( targetLocales, _mt, tlArg) )
+// where tlArg is a (Chapel) array or domain or distribution.
+static bool isResultOfTargetLocalesCall(Symbol* actual) {
+  if (! actual->hasFlag(FLAG_TEMP)) return false;
+  SymExpr* def = actual->getSingleDef();
```

Comment:
This seems like it will trigger on uses of symbols that store the result of targetLocales calls in addition to the calls themselves.  I may be misreading, though, would you mind adding a test that ensures you only get this error message once per targetLocales call in that case?

Reply:
I believe my deprecation test checks this case. It produces a single warning message, as desired.
What other cases are you thinking about?

```chpl
// here is test/deprecated/targetLocales.chpl
const D = newBlockDom({1..3, 1..4});
var A: [D] int;

writeln(A.targetLocales());
writeln(D.targetLocales());
writeln(D.dist.targetLocales());
```

Reply:
By "storing the result of [the] call", I mean something like:
```chapel
const D = newBlockDom({1..3, 1..4});
var A: [D] int;

var targetLocs = A.targetLocales();
writeln(targetLocs);
```
Ideally, we want the warning to trigger on line 4.  I'm worried this code would trigger the error message twice, since I would expect the symbol `targetLocs` to be referenced twice and the code you've written looks like it jumps to the definition point of the symbol rather than looking in the currently provided call.  Or maybe it doesn't in this situation because it's only "named" in one case but there's another situation where preFoldNamed is called where we'd generate the warning a second time for a single `targetLocales()` call.